### PR TITLE
Database update

### PR DESCRIPTION
**Pull request by Netmisa**
# Description

New model of database implemented.

### CODE REVIEW

File: app/DoctrineMigrations/Version20170830115111.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'postgresql', 'Migration can only be executed safely on \'postgresql\'.');
+
+        $this->addSql('TRUNCATE TABLE equipment, equipment_status, location_patch');
```
Comment (vincentlepot) [inline app/DoctrineMigrations/Version20170830115111.php]: Not sure it's a good practice to truncate tables in migrations. Not a big deal here, but don't take the habit.

Comment (vincentlepot) [reply app/DoctrineMigrations/Version20170830115111.php]: Moreover, you drop all 3 tables afterwards :)

Comment (Netmisa) [reply app/DoctrineMigrations/Version20170830115111.php]: If I don't do this I have an SQL error. But I will remove this and in our scenario we will do manually. Thanks

File: app/DoctrineMigrations/Version20170830115230.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
+<?php
```
Comment (vincentlepot) [inline app/DoctrineMigrations/Version20170830115230.php]: For me this migration is useless. You can remove it.

File: src/AppBundle/Entity/Equipment.php [hunk 3]
```diff
@@ -7,89 +7,54 @@
...
+     * @ORM\Column(name="source_name", type="string", length=255)
      */
-    private $status;
+    protected $sourceName;
 
     /**
      * @var \DateTime
      *
      * @Gedmo\Timestampable(on="create")
      * @ORM\Column(name="created_at", type="datetime", options={"default"="CURRENT_TIMESTAMP"})
```
Comment (vincentlepot) [inline src/AppBundle/Entity/Equipment.php]: Replace with now() to avoid a migrations loop

File: src/AppBundle/Entity/Equipment.php [hunk 4]
```diff
@@ -7,89 +7,54 @@
...
      */
-    private $createdAt;
+    protected $createdAt;
 
     /**
      * @var \DateTime
      *
      * @Gedmo\Timestampable(on="update")
      * @ORM\Column(name="updated_at", type="datetime", options={"default"="CURRENT_TIMESTAMP"})
```
Comment (vincentlepot) [inline src/AppBundle/Entity/Equipment.php]: Replace with now() to avoid a migrations loop

File: src/AppBundle/Entity/Location.php [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+     *
+     * @var User The user submitting the patch
+     */
+    private $user;
+
+    /**
+     * @var \DateTime
+     *
+     * @Gedmo\Timestampable(on="create")
+     * @ORM\Column(name="created_at", type="datetime", options={"default"="CURRENT_TIMESTAMP"})
```
Comment (vincentlepot) [inline src/AppBundle/Entity/Location.php]: Replace with now() to avoid a migrations loop

File: src/AppBundle/Entity/Location.php [hunk 6]
```diff
@@ -0,0 +1,389 @@
...
+     */
+    private $createdAt;
+
+    /**
+     * @var \DateTime
+     *
+     * @Gedmo\Timestampable(on="update")
+     * @ORM\Column(name="updated_at", type="datetime", options={"default"="CURRENT_TIMESTAMP"})
```
Comment (vincentlepot) [inline src/AppBundle/Entity/Location.php]: Replace with now() to avoid a migrations loop

**Comment by vincentlepot** [comment]
Same updates are needed in Status.php and User.php

**Comment by vincentlepot** [comment]
Instead of having 2 migration files, merge both in one file (redoing diff from the beginning of your PR)

File: app/DoctrineMigrations/Version20170830115111.php [hunk 7]
```diff
@@ -23,6 +23,8 @@ public function up(Schema $schema)
         $this->addSql('DROP SEQUENCE equipment_status_id_seq CASCADE');
         $this->addSql('CREATE SEQUENCE status_id_seq INCREMENT BY 1 MINVALUE 1 START 1');
         $this->addSql('CREATE SEQUENCE location_id_seq INCREMENT BY 1 MINVALUE 1 START 1');
+        $this->addSql('CREATE SEQUENCE elevator_id_seq INCREMENT BY 1 MINVALUE 1 START 1');
```
Comment (vincentlepot) [inline app/DoctrineMigrations/Version20170830115111.php]: There should be an annotation for that. Moreover, I don't think integer is a good option for an ID in an API like Hermod. You should consider UUID.

Comment (vincentlepot) [reply app/DoctrineMigrations/Version20170830115111.php]: (same for the other entities of course)

**Comment by vincentlepot** [comment]
Sorry for the wrong way, but we will need to revert to CURRENT_TIMESTAMP as now() is wrongly quoted by doctrine.
I'll do it if you want

**Comment by vincentlepot** [comment]
I believe you missed to add an update in docker-compose.yml (to take the uuid extension script into account)

**Pull request merged**